src/components/budget/dto/budget-record.dto.ts

```diff
@@ -17,4 +18,6 @@ export class BudgetRecord extends Resource {
 
   @Field()
   readonly amount: SecuredFloatNullable;
+
+  readonly canDelete: SecuredBoolean;
```

Comment:
canDelete is not a secured boolean. The boolean represents whether the requesting user can delete the resource or not. It doesn't make sense for the user to not be able to read it and there's nothing to edit there.

I'm assuming this is just trying to sync between java. I'd suggest dropping all the canDelete changes and additions as we haven't really worked out how they will work and they are not the same as properties.